azure Gradle plugin and host.json local.setting.json to match values in sample Azure Functions application

sdelamo opened the pull request:
see: https://plugins.gradle.org/plugin/com.microsoft.azure.azurefunctions
see: https://learn.microsoft.com/en-us/azure/azure-functions/functions-create-first-java-gradle
see: https://github.com/Azure-Samples/azure-functions-samples-java

sdelamo committed:
azure plugin [host|local.setting].json [ci skip]

see: https://plugins.gradle.org/plugin/com.microsoft.azure.azurefunctions
see: https://learn.microsoft.com/en-us/azure/azure-functions/functions-create-first-java-gradle

Updates local.settings.json
and host.json to match current value in sample app

https://github.com/Azure-Samples/azure-functions-samples-java

sdelamo commented on starter-core/src/test/groovy/io/micronaut/starter/feature/function/azure/AzureFeatureValidatorSpec.groovy:
Validator was removed https://github.com/micronaut-projects/micronaut-starter/pull/1663

timyates commented on starter-azure-function/hosts.json:
Shouldn't this be called `host.json` not `hosts.json`?

sdelamo replied:
Probably, it should be `host.json`.

sdelamo replied:
done via https://github.com/micronaut-projects/micronaut-starter/pull/1666/commits/588933906ea26589b2a81267957599091bf4c1c4

timyates commented on starter-core/src/main/resources/functions/azure/local.settings.json:
Do we need to also update `starter-azure-function/local.settings.json`?

sdelamo replied:
done via https://github.com/micronaut-projects/micronaut-starter/pull/1666/commits/588933906ea26589b2a81267957599091bf4c1c4

timyates commented:
The module `starter-azure-function` is only included in `settings.gradle` for java 8

Since https://azure.microsoft.com/en-gb/updates/java-11-support-for-azure-functions-is-now-generally-available/

Isn't Java 11 supported also?

Actually

https://azure.microsoft.com/en-gb/updates/generally-available-azure-functions-support-for-java-17/

Looks like Java 17 is also supported now?

sdelamo commented:
> The module `starter-azure-function` is only included in `settings.gradle` for java 8
> 
> Since https://azure.microsoft.com/en-gb/updates/java-11-support-for-azure-functions-is-now-generally-available/
> 
> Isn't Java 11 supported also?
> 
> Actually
> 
> https://azure.microsoft.com/en-gb/updates/generally-available-azure-functions-support-for-java-17/
> 
> Looks like Java 17 is also supported now?

Yes, they support now 17

sdelamo committed:
update files in starter-azure-function

sdelamo committed:
starter-azure-function for 11 and 17

sdelamo commented:
I have [included azure-function 11 and 17](https://github.com/micronaut-projects/micronaut-starter/pull/1666/commits/e1ab20135a8ba4a30571785c63b53ab4f8f19ffd)

timyates reviewed (approved):
[I think we need to update this to include java 17](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L117-L126)

Also, we seem to specify [windows for maven builds](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L109), but linux for [gradle builds](https://github.com/micronaut-projects/micronaut-starter/blob/e1ab20135a8ba4a30571785c63b53ab4f8f19ffd/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/template/azurefunctions.rocker.raw#L13)

Also, Maven builds have the [java version specified](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L110), but we don't do the same for Java

sdelamo committed:
Merge branch '3.9.x' into 39ximp11

sdelamo committed:
remove unused import

sdelamo committed:
Add Java 17

sdelamo committed:
azure-functions-maven-plugin to 1.24.0

https://central.sonatype.com/artifact/com.microsoft.azure/azure-functions-maven-plugin/1.24.0

sdelamo committed:
com.microsoft.azure.azurefunctions to 1.12.1

https://plugins.gradle.org/plugin/com.microsoft.azure.azurefunctions

sdelamo commented:
> [I think we need to update this to include java 17](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L117-L126)

> Also, Maven builds have the [java version specified](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L110), but we don't do the same for Java

Done for both https://github.com/micronaut-projects/micronaut-starter/pull/1666/commits/dc493ea61b047649e8209b38d0b23753afce6d38
